test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/AddressRegistrationTests.cs

```diff
@@ -40,58 +56,110 @@ public class AddressRegistrationTests
                 .UseConfiguration(config)
-                .UseKestrel()
+                .UseKestrel(options =>
+                {
+                    options.UseHttps(@"TestResources/testCert.pfx", "testPassword");
+                })
                 .Configure(ConfigureEchoAddress);
 
             using (var host = hostBuilder.Build())
             {
                 host.Start();
 
-                using (var client = new HttpClient())
+                RemoteCertificateValidationCallback validationCallback =
+                        (sender, cert, chain, sslPolicyErrors) => true;
+
+                try
                 {
-                    foreach (var testUrl in testUrls)
+#if NET451
+                    var handler = new HttpClientHandler();
+                    ServicePointManager.ServerCertificateValidationCallback += validationCallback;
+#else
+                    var handler = new WinHttpHandler();
+                    handler.ServerCertificateValidationCallback += (sender, cert, chain, sslPolicyErrors) => true;
+#endif
+
+                    using (var client = new HttpClient(handler))
                     {
-                        var responseText = await client.GetStringAsync(testUrl);
-                        Assert.Equal(testUrl, responseText);
+                        foreach (var testUrl in testUrls(host.ServerFeatures.Get<IServerAddressesFeature>()))
+                        {
+                            var responseText = await client.GetStringAsync(testUrl);
+                            Assert.Equal(testUrl, responseText);
+                        }
                     }
                 }
+                finally
+                {
+#if NET451
+                    ServicePointManager.ServerCertificateValidationCallback -= validationCallback;
+#endif
+                }
             }
         }
 
-        public static TheoryData<string, string[]> AddressRegistrationDataIPv4
+        public static TheoryData<string, Func<IServerAddressesFeature, string[]>> AddressRegistrationDataIPv4
         {
             get
             {
-                var port1 = PortManager.GetPort();
-                var port2 = PortManager.GetPort();
-                var dataset = new TheoryData<string, string[]>();
-                dataset.Add($"{port1}", new[] { $"http://localhost:{port1}/" });
-                dataset.Add($"{port1};{port2}", new[] { $"http://localhost:{port1}/", $"http://localhost:{port2}/" });
-                dataset.Add($"http://127.0.0.1:{port1}/", new[] { $"http://127.0.0.1:{port1}/", });
-                dataset.Add($"http://localhost:{port1}/base/path", new[] { $"http://localhost:{port1}/base/path" });
+                var dataset = new TheoryData<string, Func<IServerAddressesFeature, string[]>>();
+
+                // Default host and port
+                dataset.Add(null, _ => new[] { "http://localhost:5000/" });
+                dataset.Add(string.Empty, _ => new[] { "http://localhost:5000/" });
```

Comment:
I don't think this is related to my changes, since multiple theories where using the same port before my changes.  By default, tests in the same class are not run in parallel by XUnit, so these tests should be fine.